server/__test__/service/DeleteComment.test.js

```diff
@@ -0,0 +1,121 @@
+const { deleteComment } = require("../../src/service/CommentsService");
+const commentsDao = require("../../src/repository/CommentsDAO");
+const ArgumentError = require("../../src/errors/ArgumentError");
+
+describe("Test DeleteComment", ()=> {
+    //test success of the deleteComment feature
+    test("Test Success for User", async ()=> {
+        const inputData = {
+            username: "Test",
+            isAdmin: false,
+            reviewId: "1",
+            commentId: "1"
+        }
+
+        const expectedComment = {
+            author: "Test",
+            reviewId: "1",
+            commentId: "1"
+        }
+
+        jest.spyOn(commentsDao, "getOneCommentById").mockReturnValueOnce(expectedComment);
+        jest.spyOn(commentsDao, "deleteCommentById").mockReturnValueOnce(true);
+
+        const result = await deleteComment(inputData);
+
+        expect(result).toStrictEqual(expectedComment);
```

Comment:
Objects are mutable, so this will always be true.  If code within the function accidentally mutates the passed-in Object, `expectedComment` will not be what it initially is.

To fix this, you can pass in deep clones of objects by using `structuredClone()`.